packages/langium/src/grammar/type-system/type-collector/types.ts

```diff
@@ -148,12 +148,21 @@ export class InterfaceType {
     properties: Property[] = [];
 
     get superProperties(): Property[] {
+        return this.getSuperProperties(new Set());
+    }
+
+    private getSuperProperties(visited: Set<string>): Property[] {
```

Comment:
Not required in this PR necessarily, but might be a good time to add a function header detailing how this should work, and why visited is important to prevent cycles.